tests/test_rank.py

```diff
@@ -403,42 +393,59 @@ def test_get_knn_distance_ood_scores():
+    knn_distance_ood_score = rank.get_outlier_scores(features=X_test_with_ood, knn=knn, k=k)
 
     # Index of the datapoint with max ood score should correspond to the last element of X_test_with_ood
     idx_max_score = np.argsort(knn_distance_ood_score)[-1]
     assert idx_max_score == (X_test_with_ood.shape[0] - 1)
 
     # Get KNN distance as OOD score without passing k
-    # By default k=min(10, max_k) is used where max_k is extracted from the given nbrs
-    knn_distance_ood_score = rank.get_knn_distance_ood_scores(features=X_test_with_ood, nbrs=nbrs)
+    # By default k=min(10, max_k) is used where max_k is extracted from the given knn
+    knn_distance_ood_score = rank.get_outlier_scores(features=X_test_with_ood, knn=knn)
 
     # Index of the datapoint with max ood score should correspond to the last element of X_test_with_ood
     idx_max_score = np.argsort(knn_distance_ood_score)[-1]
     assert idx_max_score == (X_test_with_ood.shape[0] - 1)
 
 
-def test_bad_k_for_get_knn_distance_ood_scores():
-
-    X_train = data["X_train"]
-    X_test = data["X_test"]
+def test_default_k_and_model_get_outlier_scores():
+    # Testing using 'None' as classifier param and correct setting of default k as max_k
 
-    # Create OOD datapoint
+    # Create dataset with OOD datapoint
+    X = data["X_test"]
     X_ood = np.array([[999999999.0, 999999999.0]])
+    X_with_ood = np.vstack([X, X_ood])
 
-    # Add OOD datapoint to X_test
-    X_test_with_ood = np.vstack([X_test, X_ood])
+    instantiated_k = 10
 
-    # Fit nearest neighbors on X_train
-    nbrs = NearestNeighbors(n_neighbors=5).fit(X_train)
+    # Create NN classifiers with small instantiated k and fit on data
+    knn = NearestNeighbors(n_neighbors=instantiated_k, metric="cosine").fit(X_with_ood)
 
-    with pytest.raises(AssertionError) as e:
-        _ = rank.get_knn_distance_ood_scores(
-            features=X_test_with_ood, nbrs=nbrs, k=10  # this should throw an assertion error
+    avg_knn_distances_default_model = rank.get_outlier_scores(
+        features=X_with_ood,
+        k=instantiated_k,  # this should use default classifier (same as above) and k = instantiated_k
+    )
+
+    avg_knn_distances_default_k = rank.get_outlier_scores(
+        features=X_with_ood  # default k should be set to 10 == instantiated_k
+    )
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        # Trigger a warning.
+        avg_knn_distances = rank.get_outlier_scores(
+            features=X_with_ood,
+            knn=knn,
+            k=25,  # this should throw warn, k should be set to instantiated_k
         )
+
+        assert len(w) == 1
+        assert "Value of k changed" in str(w[-1].message)
+
+    # Score sums should be equal because the three classifiers used have identical params and fit
+    assert avg_knn_distances.sum() == avg_knn_distances_default_model.sum()
+    assert avg_knn_distances_default_k.sum() == avg_knn_distances.sum()
```

Comment:
maybe also verify the distances are **not** the same if the user specified a different value of k like k=1?